examples/tutorial/hello/go/README.md

```diff
@@ -1,66 +1,138 @@
-# Tutorial 1: Go Function w/ Input (3 minutes)
+# Go Function Hello World
 
-This example will show you how to test and deploy Go (Golang) code to Fn. It will also demonstrate passing data in through stdin.
+This example will shows you how to test and deploy Go code to Fn. It will also demonstrates passing JSON data to your function through `stdin`.
 
-### First, run the following commands:
+This tutorial assumes you have installed Docker, Fn server, and Fn CLI. See the [Fn Quickstart](https://github.com/fnproject/fn) for installation steps.
+
+## Start Fn Server
+
+Start up the Fn server so we can deploy our function.
 
 ```sh
-# Initialize your function creating a func.yaml file
-fn init --name hello-go
+fn start
+```
 
-# Test your function. This will run inside a container exactly how it will on the server
-fn run
+The command starts Fn in single server mode using an embedded database and message queue. You can find all the
+configuration options [here](docs/operating/options.md). If you are on Windows, check [here](docs/operating/windows.md).
 
-# Now try with an input
-cat sample.payload.json | fn run
+## Create your Function 
 
-# Deploy your functions to the Fn server (default localhost:8080)
-# This will create a route to your function as well
-fn deploy --app myapp
-```
+1. Create an empty directory called `hello` and `cd` into it.
+1. Create a source file for your Go function: `func.go`.
+1. Add the following code to the file.
 
-### Now call your function:
+```go
+package main
 
-```sh
-curl http://localhost:8080/r/myapp/hello-go
+import (
+	"encoding/json"
+	"fmt"
+	"os"
+)
+
+type Person struct {
+	Name string
+}
+
+func main() {
+	p := &Person{Name: "World"}
+	json.NewDecoder(os.Stdin).Decode(p)
+	mapD := map[string]string{"message": fmt.Sprintf("Hello %s!", p.Name)}
+	mapB, _ := json.Marshal(mapD)
+	fmt.Println(string(mapB))
+}
 ```
+<ol start="4">
+  <li>Initialize your function file by entering.</li>
+</ol>
 
-Or call from a browser: [http://localhost:8080/r/myapp/go](http://localhost:8080/r/myapp/hello-go)
+    fn init
 
-And now with the JSON input:
+Which returns
 
 ```sh
-curl -H "Content-Type: application/json" -X POST -d @sample.payload.json http://localhost:8080/r/myapp/hello-go
+Found go, assuming go runtime.
+func.yaml created
 ```
 
-That's it!
+Examine the files in your directory. Fn found your `func.go` file and generated the following `func.yaml` file:
+
+```yaml
+name: hello
+version: 0.0.1
+runtime: go
+entrypoint: ./func
```

Comment:
better to add explicitly `type`, `timeout`

Reply:
Is this really needed for  a Hello World example?  As a noob, can you tell more more about this or where I could read more about type and timeout?

Reply:
Take a look at this doc: https://github.com/fnproject/fn/blob/master/docs/function-file.md
I do think that we should explain what all fields mean and which ones can be omitted in favour of default values.